.github/README.md

```diff
@@ -0,0 +1,95 @@
+
+### Key Details
+
+**Naming Convention**
+
+Each step in a job has a clear name that encapsulates the purpose of the command. The convention we are using is each word in the name should be capitalized except articles and prepositions. This creates consistent labeling when looking at the progress of the current workflow on GitHub.
+
+```yaml
+- name: Checkout Repo
+# commands
+- name: Get Dependencies
+# commands
+```
+
+**Checkout Version**
+
+Current build-image ships with an older version of Git which breaks github/actions@v2 so we are using actions/checkout@v1 for all jobs until the quay image is updated to ship with a more recent version of Git.
+
+```yaml
+- name: Checkout Repo
+  uses: actions/checkout@v1
+```
+
+**Symbolic Link to Expected Workspace**
+
+A significant number of commands in the Makefile are hardcoded with an assumed file structure of the CI container. To ensure paths specified in previous commands don’t break, a symlink was created from the hardcoded “expected” working directory `/go/src/github.com/cortexproject/cortex` to the actual working directory `$GITHUB_WORKSPACE`.
+
+```yaml
+- name: Sym link expected path to github workspace
+  run: |
+    mkdir -p /go/src/github.com/cortexproject/cortex
+    ln -s $GITHUB_WORKSPACE/* /go/src/github.com/cortexproject/cortex
+```
+
+**Sharing Artifacts Between Jobs**
+
+As of October 2020, GitHub Actions do not persist between different jobs in the same workflow. Each job is run on a fresh virtual environment (https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/introduction-to-github-actions#runners). As such, we need to upload and download artifacts to share data between jobs.
+
+| Artifact                      | Stored In | Used By                                     | Purpose of Storing Artifact |
+|-------------------------------|-----------|---------------------------------------------|-----------------------------|
+| website public                | build     | deploy_website                              | share data between jobs     |
+| Docker Images                 | build     | deploy, integration, integrations-config-db | share data between jobs     |
+| Frontend Protobuf             | build     |                                             | long term storage           |
+| Caching Index Client Protobuf | build     |                                             | long term storage           |
+| Ring Protobuf                 | build     |                                             | long term storage           |
+| Rules Protobuf                | build     |                                             | long term storage           |
+
+*Note:* Docker Images are zipped before uploading as a workaround. The images contain characters that are illegal in the upload-artifact action.
+```yaml
+- name: Compressing Images
+        run: tar -zcvf images.tar.gz /tmp/images
```

Comment:
Not sure if compression gives us anything here. We can use `tar` without `-z` flag to make it run faster (but use more disk space).

Reply:
The compression was in fact unnecessary. We removed it in the subsequent migration PR: https://github.com/cortexproject/cortex/pull/3341